conans/test/functional/build_helpers/cmake_targets_test.py

```diff
@@ -104,7 +118,11 @@ def header_only_test(self):
 
             client.run('install . %s -s build_type=Debug -g cmake_multi' % debug_install)
             client.run('install . %s -s build_type=Release -g cmake_multi' % release_install)
-            client.runner("cmake .", cwd=client.current_folder)
+
+            settings.compiler.version = "14"
+            with environment_append(vcvars_dict(settings, output=client.out)):
```

Comment:
I think that cmake doesn't take the visual studio version from the environment, I have tried in the VS prompt, and it ignores it, it takes the latest VS detected (in the registry it seems: https://gitlab.kitware.com/cmake/cmake/issues/16339)

It is weird, locally in my machine there are more tests failings. The thing is:
- Our detection finds VS 2017 (15)
-  ``cmake .`` uses as default VS 2015 (14)

And it seems right to fail, if that test fail, why the other tests don't fail?

Reply:
I would say not many tests use directly CMake and at the same time changing the input settings